plugins/aep-consent-configuration/index.ts

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+import { Event } from '@adobe/griffon-toolkit-common';
+import { SharedStateVersions } from '@adobe/griffon-toolkit-aep-mobile';
+import { SharedStateConfig } from '@adobe/griffon-toolkit-aep-mobile';
+import { ValidationPluginResult } from '../../types/validationPlugin';
+
+(function (events: Event[]): ValidationPluginResult {
+  const { toolkit: kit } = window.griffon;
+  const { sharedStateVersions: versions } = kit['aep-mobile'];
+  const { sharedStateConfig: config } = kit['aep-mobile'];
+  const versionEvents = kit.match(
+    versions.matcher,
+    events
+  ) as SharedStateVersions[]
+  const configEvents = kit.match(
+    config.matcher,
+    events
+  ) as SharedStateConfig[];
+
+  const consentSDKRegistered= versionEvents.some(versions.getExtensionsKey('"com.adobe.edge.consent"'));
+
+  const consentDefaultEvents = kit.match('payload.metadata."state.data"."consent.default"', configEvents);
+
+  const consentTagInstalled = kit.search('payload.metadata."state.data"."consent.default"' , configEvents);
```

Comment:
I don't think this is doing what you want. JMESPath search assumes a JSON object, not an array. 

There's multiple ways of finding the consent collect val.

You can still use `search` to search all matched events. Just change the path to start with `[*]`. Ex: `[*].payload.metadata."state.data"."consent.default.consents.collect.val"`

Note that this method will return the values and not the entire event object. Ex: `['y', 'y', 'n']`

Or you can call `match` with your own matcher expression:

```
const myMatcher = kit.match(
  `payload.metadata."state.data"."consent.default".consents.collect.val == 'y'`
);
```

Or you can iterate through the matched events and do something like:
```
const atLeastOneY = configEvents.some(event => event.payload.metadata['state.data']['consent.default'].consents.collect.val === 'y');
```